src/FollowMe/FollowMe.cc

```diff
@@ -0,0 +1,218 @@
+        _gcsMotionReportTimer.stop();
+    }
+}
+
+void FollowMe::_setGPSLocation(QGeoPositionInfo geoPositionInfo)
+{
+    if (geoPositionInfo.isValid())
+    {
+        // get the current location coordinates
+
+        QGeoCoordinate geoCoordinate = geoPositionInfo.coordinate();
+
+        _motionReport.lat_int = geoCoordinate.latitude()*1e7;
+        _motionReport.lon_int = geoCoordinate.longitude()*1e7;
+        _motionReport.alt = geoCoordinate.altitude();
+
+#ifdef QT_QML_DEBUG
+        if(_simulate_motion == true) {
+            _motionReport.lat_int = 47.3977420*1e7;
+            _motionReport.lon_int = 8.5455941*1e7;
+            _motionReport.alt = 488.00;
+        }
+#endif
+        // get the current eph and epv
+
+        if(geoPositionInfo.hasAttribute(QGeoPositionInfo::HorizontalAccuracy) == true) {
+            _motionReport.pos_std_dev[0] = _motionReport.pos_std_dev[1] = geoPositionInfo.attribute(QGeoPositionInfo::HorizontalAccuracy);
+        }
+
+        if(geoPositionInfo.hasAttribute(QGeoPositionInfo::VerticalAccuracy) == true) {
+            _motionReport.pos_std_dev[2] = geoPositionInfo.attribute(QGeoPositionInfo::VerticalAccuracy);
+        }
+
+        // calculate z velocity if it's availible
+
+        if(geoPositionInfo.hasAttribute(QGeoPositionInfo::VerticalSpeed)) {
+            _motionReport.vz = geoPositionInfo.attribute(QGeoPositionInfo::VerticalSpeed);
+        }
+
+        // calculate x,y velocity if it's availible
+
+        if((geoPositionInfo.hasAttribute(QGeoPositionInfo::Direction)   == true) &&
+           (geoPositionInfo.hasAttribute(QGeoPositionInfo::GroundSpeed) == true)) {
+
+            qreal direction = _degreesToRadian(geoPositionInfo.attribute(QGeoPositionInfo::Direction));
+            qreal velocity = geoPositionInfo.attribute(QGeoPositionInfo::GroundSpeed);
+
+            _motionReport.vx = cos(direction)*velocity;
+            _motionReport.vy = sin(direction)*velocity;
+        }
+    }
+}
+
+void FollowMe::_sendGCSMotionReport(void)
+{
+    QmlObjectListModel & vehicles = *_toolbox->multiVehicleManager()->vehicles();
+    MAVLinkProtocol* mavlinkProtocol = _toolbox->mavlinkProtocol();
+    mavlink_follow_target_t follow_target;
+
+    memset(&follow_target, 0, sizeof(mavlink_follow_target_t));
+
+    follow_target.timestamp = runTime.nsecsElapsed()*1e-6;
+    follow_target.est_capabilities = (1 << POS);
+    follow_target.position_cov[0] = _motionReport.pos_std_dev[0];
+    follow_target.position_cov[2] = _motionReport.pos_std_dev[2];
+    follow_target.alt = _motionReport.alt;
+    follow_target.lat = _motionReport.lat_int;
+    follow_target.lon = _motionReport.lon_int;
+
+#ifdef QT_QML_DEBUG
+    if(_simulate_motion == true) {
+        _createSimulatedMotion(follow_target);
+    }
+#endif
+
+    for (int i=0; i< vehicles.count(); i++) {
+        Vehicle* vehicle = qobject_cast<Vehicle*>(vehicles[i]);
+        if(vehicle->flightMode().compare(_followMeStr, Qt::CaseInsensitive) == 0) {
```

Comment:
hey @DonLakeFlyer I'm still confused as to why this is not a generic message.  Let me know when you plan to discuss and Ill try and be there.  I can help integrate this too if you want to give me an outline of what needs to be done I can make the changes.  I'm planning on making a lot more improvements to the tracking ability of the uav through qgc so the more I understand the direction of the architecture towards APM and PX4 the better.